sdk-node/src/workflows/workflow.ts

```diff
@@ -121,43 +129,52 @@ export class Workflow<TInput extends WorkflowInput, name extends string> {
               ? `${executionId}.${config.name}.${config.runId}`
               : `${executionId}.${config.name}.${cyrb53(
                   JSON.stringify([
-                    config.systemPrompt,
-                    resultSchema,
+                    config.facts,
+                    config.goals,
                     config.input,
+                    executionId,
+                    resultSchema,
                     this.name,
                     version,
                   ]),
                 )}`;
 
-            console.log("---  Creating run", { runId, name: config.name });
-
             const result = await this.inferable.getClient().createRun({
               params: {
                 clusterId: await this.inferable.getClusterId(),
               },
               body: {
                 id: runId,
-                systemPrompt: config.systemPrompt,
+                systemPrompt: [
+                  "You are a helpful assistant.",
+                  "# Facts:",
+                  ...config.facts.map((f) => {
+                    if (typeof f === "string") {
+                      return `- ${f}`;
+                    } else {
+                      return `- ${f.description}: ${JSON.stringify(f.data)}`;
+                    }
+                  }),
+                ].join("\n"),
                 resultSchema,
-                context: config.input ? { input: config.input } : undefined,
                 onStatusChange: {
                   type: "workflow",
                   statuses: ["failed", "done"],
                   workflow: {
                     executionId: executionId,
                   },
                 },
-                initialPrompt: JSON.stringify(config.input),
+                context: {
+                  input: config.input,
```

Comment:
How come the input get's passed into the Run's context?
It should be possible to set these independently.

Reply:
Yep agree. Just an oversight on my part.